src/server/pps/server/s3g_sidecar_proxy.go

```diff
@@ -0,0 +1,283 @@
+					if p != ret {
+						logrus.Infof("PROXY reverse transformed '%s' to '%s'", p, ret)
+					}
+				} else {
+					if len(s) <= 100 {
+						logrus.Infof("PROXY Warning! LastSeenPathToReplace was empty when unmashing '%s'", s)
+					}
+				}
+				// logrus.Infof("MASHUP '%s' ==> '%s'", s, ret)
+				return ret
+			}
+
+			// logrus.Infof("===> IN PROXY ROUTER, w=%+v, r=%+v", w, r)
+
+			proxy := &httputil.ReverseProxy{
+				Director: func(req *http.Request) {
+					u := os.Getenv("MINIO_ENDPOINT")
+					// TODO: check MINIO_SECURE
+					// TODO: use MINIO_SECRET, MINIO_ID, ... need to sign
+					// (after unwrapping??) the request before we send it
+					// onwards: see https://github.com/smarty-archives/go-aws-auth (deprecated!) etc
+					// TODO: need to add r.URL.Path?
+					logrus.Infof("PROXY: r.URL.Path=%s", r.URL.Path)
+					target, err := url.Parse(fmt.Sprintf("http://%s", u))
+					if err != nil {
+						log.Fatal(err)
+					}
+					targetQuery := target.RawQuery
+					req.URL.Scheme = target.Scheme
+					req.URL.Host = target.Host
+					req.URL.Path, req.URL.RawPath = joinURLPath(target, req.URL)
+					if targetQuery == "" || req.URL.RawQuery == "" {
+						req.URL.RawQuery = targetQuery + req.URL.RawQuery
+					} else {
+						req.URL.RawQuery = targetQuery + "&" + req.URL.RawQuery
+					}
+					req.URL.Path = mashup(req.URL.Path)
+					req.URL.RawPath = mashup(req.URL.RawPath)
+					req.URL.RawQuery = mashup(req.URL.RawQuery)
+
+					// mashup each of the request headers
+					for k, v := range req.Header {
+						for i, vv := range v {
+							v[i] = mashup(vv)
+						}
+						req.Header[k] = v
+					}
+
+					if _, ok := req.Header["User-Agent"]; !ok {
+						// explicitly disable User-Agent so it's not set to default value ???
+						req.Header.Set("User-Agent", "")
+					}
+					// // If calling the istio ingress, we need to set the endpoint host in the header
+					// if coords.Host != "" {
+					// 	// req.Header.Set("Host", coords.Host)
+					// 	req.Host = coords.Host
+					// }
+
+					if req.Body != nil {
+						bodyBytes, err := ioutil.ReadAll(req.Body)
+						if err != nil {
+							logrus.Fatal(err)
+						}
+						// TODO find and replace
+						mashed := mashup(string(bodyBytes))
+						modifiedBodyBytes := new(bytes.Buffer)
+						modifiedBodyBytes.WriteString(mashed)
+						req.Body = ioutil.NopCloser(modifiedBodyBytes)
+						req.ContentLength = int64(len(mashed))
+						// prev := req.Header.Get("Content-Length")
+						// req.Header.Set("Content-Length", fmt.Sprintf("%d", len(mashed)))
+						// logrus.Infof("PROXY request switching Content-Length from %d to %d", prev, len(mashed))
+					}
+					// TODO: check whether awsauth.Sign4 reads the whole request
+					// body into memory, if it does that's bad for large writes
+					// and we should figure out how we can stream it to disk
+					// first...
+					awsauth.Sign4(req, awsauth.Credentials{
```

Comment:
Looks like we probably need to plumb the volume with the web identity token into it. See https://pachyderm.slack.com/archives/C01LBA4NSJU/p1666033437705929?thread_ts=1666032276.294569&cid=C01LBA4NSJU